Packs/Rapid7_Nexpose/Integrations/Rapid7_Nexpose/Rapid7_Nexpose.py

```diff
@@ -148,12 +150,24 @@ def __init__(self, url: str, username: str, password: str, token: str | None = N
 
     def _http_request(self, **kwargs):
         """Wrapper for BaseClient._http_request() that optionally removes `links` keys from responses."""
-        response = super()._http_request(**kwargs)
-
-        if REMOVE_RESPONSE_LINKS:
-            return remove_dict_key(response, "links")
+        for _time in range(1, CONNECTION_ERRORS_RETRIES + 1):
+            try:
+                response = super()._http_request(**kwargs)
+                if REMOVE_RESPONSE_LINKS:
+                    return remove_dict_key(response, "links")
+                return response
+            except (DemistoException, requests.ReadTimeout) as error:
```

Comment:
You have here `except (DemistoException, requests.ReadTimeout) as error`
and the condition in line 163 `not isinstance(error.exception, requests.ConnectionError)`
will always be true because you don't catch `requests.ConnectionError`

Reply:
yeah so its a bit confusing, but the `_http_request` of `BaseClient` actually catches connection errors and embeds them with `DemistoException`, however `ReadTimeout` errors are not getting caught in `_http_request` of `BaseClient`, hence what you say here is not true.